File: scripts/agent-bench/fixture-swift/Sources/Corpus/Rank.swift
public let RRF_K: Float = 60.0

public func blendScores(_ lexical: [(String, Float)], _ semantic: [(String, Float)]) -> [(String, Float)] {
    var fused: [String: Float] = [:]

    for (rank, entry) in lexical.enumerated() {
        fused[entry.0, default: 0.0] += 1.0 / (RRF_K + Float(rank) + 1.0)
    }
    for (rank, entry) in semantic.enumerated() {
        fused[entry.0, default: 0.0] += 1.0 / (RRF_K + Float(rank) + 1.0)
    }

    return fused
        .map { ($0.key, $0.value) }
        .sorted { $0.1 > $1.1 }
}

File: scripts/agent-bench/fixture-php/src/Rank.php
<?php

declare(strict_types=1);

namespace Corpus;

class Rank
{
    public const RRF_K = 60.0;

    /**
     * Reciprocal-rank fusion of two ranked (id, score) lists.
     *
     * Each list contributes 1 / (RRF_K + rank + 1) to a document's fused
     * score, where rank is the document's zero-based position in that list.
     * The result is sorted by fused score, descending.
     *
     * @param list<array{0: int, 1: float}> $lexical
     * @param list<array{0: int, 1: float}> $semantic
     * @return list<array{0: int, 1: float}>
     */
    public static function blendScores(array $lexical, array $semantic): array
    {
        /** @var array<int, float> $fused */
        $fused = [];

        foreach ([$lexical, $semantic] as $ranking) {
            foreach ($ranking as $rank => $pair) {
                $id = $pair[0];
                $contribution = 1.0 / (self::RRF_K + $rank + 1);
                $fused[$id] = ($fused[$id] ?? 0.0) + $contribution;
            }
        }

        $results = [];
        foreach ($fused as $id => $score) {
            $results[] = [$id, $score];
        }

        usort($results, static fn (array $a, array $b): int => $b[1] <=> $a[1]);

        return $results;
    }
}
